src/Compilers/Core/MSBuildTask/BuildClient.cs

```diff
@@ -31,11 +31,17 @@ internal static class BuildClient
         private const int TimeOutMsNewProcess = 20000; 
 
         /// <summary>
-        /// Try to get the directory this assembly is in.
+        /// Try to get the directory this assembly is in. Returns null if assembly
+        /// was in the GAC.
         /// </summary>
-        private static string GetClientDir()
+        private static string TryGetClientDir()
         {
-            var uri = new Uri(Assembly.GetExecutingAssembly().CodeBase);
+            var assembly = Assembly.GetExecutingAssembly();
```

Comment:
There is a note on [MSDN for GetExecutingAssembly](https://msdn.microsoft.com/en-us/library/system.reflection.assembly.getexecutingassembly(v=vs.110).aspx) that says for performance reasons you should prefer to use a type from the assembly. i.e. ```typeof(BuildClient).Assembly```